### PR TITLE
BVAL-596 Move ValidateUnwrappedValue to javax.validation.metadata

### PR DESCRIPTION
**Pull request by gsmet**
* https://hibernate.atlassian.net/browse/BVAL-596

### CODE REVIEW

**Comment by gunnarmorling** [comment]
Isn't there any include of that file that needs adapting?

**Comment by gsmet** [comment]
@gunnarmorling nope, it's not described in the spec AFAICS. Maybe it should be?

**Comment by gunnarmorling** [comment]
Yes, absolutely. Each API type should be contained in the spec.

**Comment by gsmet** [comment]
@gunnarmorling I added one more commit to mention `validateUnwrappedValue()`.

File: sources/constraint-metadata.asciidoc [hunk 1]
```diff
@@ -436,6 +436,13 @@ include::{validation-api-source-dir}javax/validation/metadata/ConstraintDescript
 
 [tck-testable]#[methodname]`getConstraintValidatorClasses()` returns the [classname]`ConstraintValidator` classes associated with the constraint.#
 
+[tck-testable]#[methodname]`validateUnwrappedValue()` returns a [classname]`ValidateUnwrappedValue` value describing the unwrapping behavior.#
```
Comment (gunnarmorling) [inline sources/constraint-metadata.asciidoc]: perhaps s/value describing/instance describing/?

Comment (gsmet) [inline sources/constraint-metadata.asciidoc]: Ah, yes, it's better. Force pushed.

**Comment by gunnarmorling** [comment]
Thanks, @gsmet!

**Pull request merged**